sample.py

```diff
@@ -0,0 +1,219 @@
+import argparse
+import pickle
+import numpy as np
+import utils
+import enum
+
+import common
+from short_text_codec import ShortTextCodec
+
+MAX_PROG_SAMPLE_INTERVAL = 10000
+
+class VisInit(enum.Enum):
+    """Ways of initializing visible units before repeated gibbs sampling."""
+    zeros = 1
+    # Treat visible biases as softmax
+    biases = 2
+    # Turn on each unit (not just each one-hot vector) with p=.5
+    uniform = 3
+    spaces = 4
+    padding = 7 # Old models use ' ' as filler, making this identical to the above
+    # Training examples
+    train = 5
+    # Choose a random length. Fill in that many uniformly random chars. Fill the rest with padding character.
+    chunks = 6
+    # TODO: Use training examples but randomly mutate non-space/padding characters
+
+def print_samples(model, visibles):
+    for v in visibles:
+        print model.codec.decode(v, pretty=True)
+
+class BadInitMethodException(Exception):
+    pass
+
+def starting_visible_configs(init_method, n, model, training_examples):
+    vis_shape = (n, model.intercept_visible_.shape[0])
+    maxlen, nchars = model.codec.maxlen, model.codec.nchars
+    if init_method == VisInit.biases:
+        sm = np.tile(model.intercept_visible_, [n, 1]).reshape( (-1,) + model.codec.shape() )
+        return utils.softmax_and_sample(sm).reshape(vis_shape)
+    elif init_method == VisInit.zeros:
+        return np.zeros(vis_shape)
+    elif init_method == VisInit.uniform:
+        return np.random.randint(0, 2, vis_shape)
+    # This will fail if ' ' isn't in the alphabet of this model
+    elif init_method == VisInit.spaces or init_method == VisInit.padding:
+        fillchar = {VisInit.spaces: ' ', VisInit.padding: model.codec.filler}[init_method]
+        vis = np.zeros( (n,) + model.codec.shape())
+        try:
+            fill = model.codec.char_lookup[fillchar]
+        except KeyError:
+            raise BadInitMethodException(fillchar + " is not in model alphabet")
```

Comment:
Nit: you probably want repr(fillchar) to avoid the message "  is not in model alphabet"